src/terrain/routes/analyses.clj

```diff
@@ -13,43 +24,166 @@
 (s/defschema DeletionResponse
   {:id (describe UUID "The UUID of the resource that was deleted")})
 
-(defn secured-analyses-routes
+(defn secured-quicklaunch-routes
   "The routes for accessing analyses information. Currently limited to Quick
    Launches."
   []
   (optional-routes [config/app-routes-enabled])
 
   (context "/quicklaunches" []
-    :tags ["analyses"]
+    :tags ["analyses-quicklaunches"]
 
     (GET "/:id" [id]
-      :summary "Get Quick Launch information by its UUID."
+      :summary     "Get Quick Launch information by its UUID."
       :description "Gets Quick Launch information, including the UUID, the name of
       the user that owns it, and the submission JSON"
       :return  QuickLaunch
       (ok (analyses/get-quicklaunch id)))
 
     (PATCH "/:id" [id]
-      :body         [quicklaunch UpdateQuickLaunch]
-      :return       QuickLaunch
-      :summary      "Modifies an existing Quick Launch"
-      :description  "Modifies an existing Quick Launch, allowing the caller to change
+      :body        [quicklaunch UpdateQuickLaunch]
+      :return      QuickLaunch
+      :summary     "Modifies an existing Quick Launch"
+      :description "Modifies an existing Quick Launch, allowing the caller to change
       owners and the contents of the submission JSON"
       (ok (analyses/update-quicklaunch id quicklaunch)))
 
     (POST "/" []
-      :body         [quicklaunch NewQuickLaunch]
-      :return       QuickLaunch
-      :summary      "Adds a Quick Launch to the database"
-      :description  "Adds a Quick Launch and corresponding submission information to the
+      :body        [quicklaunch NewQuickLaunch]
+      :return      QuickLaunch
+      :summary     "Adds a Quick Launch to the database"
+      :description "Adds a Quick Launch and corresponding submission information to the
       database. The username passed in should already exist. A new UUID will be
       assigned and returned."
       (ok (analyses/add-quicklaunch quicklaunch)))
 
     (DELETE "/:id" [id]
-      :return        DeletionResponse
-      :summary      "Deletes a Quick Launch"
-      :description  "Deletes a Quick Launch from the database. Will returns a success
+      :return      DeletionResponse
+      :summary     "Deletes a Quick Launch"
+      :description "Deletes a Quick Launch from the database. Will returns a success
       even if called on a Quick Launch that has either already been deleted or never
       existed in the first place"
       (ok (analyses/delete-quicklaunch id)))))
+
+(defn secured-quicklaunch-favorites-routes
+  []
+  (optional-routes [config/app-routes-enabled])
+
+  (context "/quicklaunch/favorites" []
+    :tags ["analyses-quicklaunch-favorites"]
+
+    (GET "/:id" [id]
```

Comment:
Usually endpoints with an ID in the path should go after other endpoints with the same prefix. It doesn't seem to be an issue with these endpoints, but if we ever add a `/quicklaunch/favorites/foo` endpoint, it will have to go before this one with the ID, otherwise it will match this endpoint first and assume `foo` is an ID.

By the way, if these really are UUIDs, then adding `:path-params` to the endpoints can make them validate the UUIDs. And another reason to keep paths with IDs grouped together is that they can be under the same context where the `path-param` is defined once. For example:
https://github.com/cyverse-de/terrain/blob/cd6106ba0652acc4fc6f1d6daa8a9eaeddd5a68e/src/terrain/routes/favorites.clj#L16-L22

Reply:
Changing to what you linked to here makes sense to me, I'll see about getting the code refactored in terrain for now.